docs/decisions/0006-handling-learner-initiated-unenrollments.rst

```diff
@@ -0,0 +1,100 @@
+0006 handling learner initiated unenrollments
+#############################################
+
+Status
+******
+
+**Draft** May 2023
+
+Context
+*******
+
+The enterprise-subsidy service needs to maintain records of transaction reversals for any associated platform
+fulfillment cancellation. These reversals represent credit, up to the amount allocated by the original transaction,
+returned to the learner's customer's ledger. As there are many sources of a learner initiated unenrollments/
+cancellations, the enterprise subsidy service must take on the responsibility of monitoring the state of enterprise
+fulfillments. This is necessary to ensure that any learner initiated unenrollment or cancellation of an enterprise 
+fulfillment object will be evaluated and that corresponding reversals are written if/when applicable. 
+
+Decision
+********
+
+In a pre-event bus driven architectural world, the enterprise subsidy service will monitor the state of recently 
+modified, unenrolled enterprise fulfillment objects. It will do so by running a management command which
+will hit the platform enterprise fulfillment `API 
+<https://github.com/openedx/edx-enterprise/blob/master/enterprise/api/v1/views.py#L576>`_ somewhere between once and
+twice a day. The view will return a list of enterprise fulfillment objects which have been cancelled after the provided
+modified at timestamp filter. Notably, we provide a modified at time window that is smaller than the frequency at which
```

Comment:
I did just notice the timedelta in the code is set to 24 hours, but still, jenkins or whatever runs the management command probably can't guarantee a consistent schedule and could result in a 25+ hour period between runs.  Shouldn't we update the window to something like 7 days to give us some flexibility to recover a stuck job?

Reply:
I apologize, the wording on the ADR was out of date- we have revised our approach here to:

1) pull unenrolled_at instead of modified at on the enterprise enrollment. This is a bit safer of a value because it is _**only**_ written at the time of unenrollment. 

2) The window size (size of the `unenrolled_after` query param used by the subsidy service) of fetched unenrollments will be roughly 2x the pulling frequency (The rate at which the job runs). In this case, the the unenrolled_after would be 24 hours and the job frequency would be set to ~12 hours. This could of course change and I'm open to a conversation but I do believe that a faster frequency is desirable here as we want refunds to be realized relatively quickly. So the question then I guess is what will the `unenrolled_after` value ge? 🤔

Reply:
I'd still advocate for a 7-day window regardless of polling frequency because if the job scheduler (argocd) gets stuck for some reason, we have ~6 days to un-stuck it before we have to manually intervene to expand the window.  Are we concerned that performance won't be up to snuff for this large of a window?

Reply:
sure! Let's start with a 7 day window and we can tweak from there. I don't think that it should cause much of an issue performance wise